packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -548,9 +548,17 @@ export default class NewVm extends BaseComponent {
     const defaultNetworkIds = this._getDefaultNetworkIds(template)
     forEach(template.VIFs, vifId => {
       const vif = getObject(storeState, vifId, resourceSet)
-      VIFs.push({
-        network: pool || isInResourceSet(vif.$network) ? vif.$network : defaultNetworkIds[0],
-      })
+      // insertion of the vif at the index that corresponds to the order chosen by the user when creating the VM
+      // this order is stored in the key named "device" of the VIF object.
+      // if the index is NaN, the insertion is done at the end of the array.
+      const vifIndex = Number(vif.device)
+      if (isNaN(vifIndex)) {
+        VIFs.push({
+          network: pool || isInResourceSet(vif.$network) ? vif.$network : defaultNetworkIds[0],
+        })
+      } else {
+        VIFs.splice(vifIndex, 0, {network: pool || isInResourceSet(vif.$network) ? vif.$network : defaultNetworkIds[0]})
+      }
```

Comment:
To avoid duplication, create the object before the `if/else`.
```js
const networkObject = {
  network: pool || isInResourceSet(vif.$network) ? vif.$network : defaultNetworkIds[0]
}

if(isNan(vifIndex)){
VIFs.push(networkObject)
} else {
VIFs.splice(vifIndex, 0, networkObject)
}
```